### PR TITLE
CSPL 1218: App framework authentication tests

### PR DESCRIPTION
**Pull request by tpereirasplunk**
a test case with the following -

- Use random credentials
- Attempt Install an app from remote store, expect it to fail
- Change remote store credentials  to a new valid set
- Install a new app or update existing app from remote store, expect it to pass

Clean Run: https://github.com/splunk/splunk-operator/runs/6103552270?check_suite_focus=true

### CODE REVIEW

File: test/s1/appframework/appframework_test.go [hunk 1]
```diff
@@ -1345,4 +1345,149 @@ var _ = Describe("s1appfw test", func() {
 
 		})
 	})
+
+	Context("Standalone deployment (S1) with App Framework", func() {
+		It("smoke, s1, appframeworks1, appframework: can deploy a Standalone instance with App Framework enabled, attempt to update using incorrect S3 credentials", func() {
```
Comment (jambrosiano) [inline test/s1/appframework/appframework_test.go:1350]: Do we keep this as smoke or move it to integration?

File: test/s1/appframework/appframework_test.go [hunk 2]
```diff
@@ -1345,4 +1345,149 @@ var _ = Describe("s1appfw test", func() {
...
+			splunkPodAge := testenv.GetPodsStartTime(testcaseEnvInst.GetName())
+
+			// ############ INITIAL VERIFICATION ###########
+			standalonePod := []string{fmt.Sprintf(testenv.StandalonePod, deployment.GetName(), 0)}
+			standaloneAppSourceInfo := testenv.AppSourceInfo{CrKind: standalone.Kind, CrName: standalone.Name, CrAppSourceName: appSourceName, CrPod: standalonePod, CrAppVersion: appVersion, CrAppScope: enterpriseApi.ScopeLocal, CrAppList: appListV1, CrAppFileList: appFileList}
+			allAppSourceInfo := []testenv.AppSourceInfo{standaloneAppSourceInfo}
+			testenv.AppFrameWorkVerifications(ctx, deployment, testcaseEnvInst, allAppSourceInfo, splunkPodAge, "")
+
+			// ############  Modify secret key ###########
+			// Create App framework volume with random credentials and apply to Spec
```
Comment (jambrosiano) [inline test/s1/appframework/appframework_test.go:1431]: nit: this line says 'random' the one after says 'invalid', better say 'invalid' everywhere

File: test/s1/appframework/appframework_test.go [hunk 3]
```diff
@@ -1345,4 +1345,149 @@ var _ = Describe("s1appfw test", func() {
...
+			uploadedApps = append(uploadedApps, uploadedFiles...)
+
+			// Check for changes in App phase to determine if next poll has been triggered
+			testenv.WaitforPhaseChange(ctx, deployment, testcaseEnvInst, deployment.GetName(), standalone.Kind, appSourceName, appFileList)
+
+			//Check no apps are updated as auth key is incorrect
+			testenv.AppFrameWorkVerifications(ctx, deployment, testcaseEnvInst, allAppSourceInfo, splunkPodAge, "")
+
+			// ############  Modify secret key to correct one###########
+			//Apply spec with correct credentails
```
Comment (jambrosiano) [inline test/s1/appframework/appframework_test.go:1465]: nit: typo "credentails"

**Review by jambrosiano — APPROVED** [review]

**Review by smohan-splunk — APPROVED** [review]

File: test/s1/appframework/appframework_test.go [hunk 4]
```diff
@@ -1345,4 +1345,148 @@ var _ = Describe("s1appfw test", func() {
...
+			testenv.AppFrameWorkVerifications(ctx, deployment, testcaseEnvInst, allAppSourceInfo, splunkPodAge, "")
+
+			// ############  Modify secret key ###########
+			// Create App framework volume with invalid credentials and apply to Spec
+			testcaseEnvInst.Log.Info("Update Standalone spec with invalid credentials")
+			standalone = &enterpriseApi.Standalone{}
+			err = deployment.GetInstance(ctx, deployment.GetName(), standalone)
+			Expect(err).To(Succeed(), "Failed to get instance of Standalone")
+
+			standalone.Spec.AppFrameworkConfig.VolList[0].SecretRef = testenv.RandomDNSName(5)
```
Comment (pdhanoyasplunk) [inline test/s1/appframework/appframework_test.go:1437]: Passing incorrect secret object name is not the aim of this test. We need correct(which exists) secret object but incorrect values in that secret object. 
Ideally we should create a secret object with random keys use that in the CR.

Comment (tpereirasplunk) [reply test/s1/appframework/appframework_test.go]: modified to use secret object with random values for keys

File: test/s1/appframework/appframework_test.go [hunk 5]
```diff
@@ -1345,4 +1345,140 @@ var _ = Describe("s1appfw test", func() {
...
+			uploadedApps = append(uploadedApps, uploadedFiles...)
+
+			// Check for changes in App phase to determine if next poll has been triggered
+			testenv.WaitforPhaseChange(ctx, deployment, testcaseEnvInst, deployment.GetName(), standalone.Kind, appSourceName, appFileList)
+
+			//Check no apps are updated as auth key is incorrect
+			testenv.AppFrameWorkVerifications(ctx, deployment, testcaseEnvInst, allAppSourceInfo, splunkPodAge, "")
+
+			// ############  Modify secret key to correct one###########
+			//Apply spec with correct credentials
```
Comment (pdhanoyasplunk) [inline test/s1/appframework/appframework_test.go:1462]: Nit: Space after //

Comment (pdhanoyasplunk) [inline test/s1/appframework/appframework_test.go:1462]: Nit: Space after //

**Review by pdhanoyasplunk — APPROVED** [review]

**Pull request merged**